src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/typeresolution.cpp

```diff
@@ -0,0 +1,657 @@
+                pMetaDataDispenser,
+                pszCandidateFilePath,
+                &spMetaDataImport);
+        }
+        else
+        {
+            hr = E_OUTOFMEMORY;
+        }
+
+        if (SUCCEEDED(hr))
+        {
+            const size_t cFullName = wcslen(pszFullName);
+            wchar_t pszRetrievedName[g_uiMaxTypeName];
+            HCORENUM hEnum = nullptr;
+            mdTypeDef rgTypeDefs[32];
+            ULONG cTypeDefs;
+            DWORD dwTypeDefProps;
+            hr = RO_E_METADATA_NAME_NOT_FOUND;
+
+            if (TRO_RESOLVE_TYPE & resolutionOptions)
+            {
+                hr = spMetaDataImport->FindTypeDefByName(pszFullName, mdTokenNil, &rgTypeDefs[0]);
+                if (SUCCEEDED(hr))
+                {
+                    //  Check to confirm that the type we just found is a
+                    //  winrt type.  If it is, we're good, otherwise we
+                    //  want to fail with RO_E_INVALID_METADATA_FILE.
+                    hr = spMetaDataImport->GetTypeDefProps(rgTypeDefs[0], nullptr, 0, nullptr, &dwTypeDefProps, nullptr);
+                    if (SUCCEEDED(hr))
+                    {
+                        //  If we found the type but it's not a winrt type,
+                        //  it's an error.
+                        //
+                        //  If the type is public, than the metadata file
+                        //  is corrupt (all public types in a winrt file
+                        //  must be tdWindowsRuntime).  If the type is
+                        //  private, then we just want to report that the
+                        //  type wasn't found.
+                        if (!IsTdWindowsRuntime(dwTypeDefProps))
+                        {
+                            if (IsTdPublic(dwTypeDefProps))
+                            {
+                                hr = RO_E_INVALID_METADATA_FILE;
+                            }
+                            else
+                            {
+                                hr = RO_E_METADATA_NAME_NOT_FOUND;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        hr = RO_E_INVALID_METADATA_FILE;
+                    }
+                    if (SUCCEEDED(hr))
+                    {
+                        if (pmdTypeDef != nullptr)
+                        {
+                            *pmdTypeDef = rgTypeDefs[0];
+                        }
+                        if (ppMetaDataImport != nullptr)
+                        {
+                            *ppMetaDataImport = spMetaDataImport.Detach();
+                        }
+                    }
+                }
+                else if (hr == CLDB_E_RECORD_NOTFOUND)
+                {
+                    hr = RO_E_METADATA_NAME_NOT_FOUND;
+                }
+            }
+
+            if ((hr == RO_E_METADATA_NAME_NOT_FOUND) &&
```

Comment:
Is the entire point of this block to selectively update the RO_E_METADATA_NAME_NOT_FOUND result to RO_E_METADATA_NAME_IS_NAMESPACE?  No issue with it, just making sure I understand correctly.

Reply:
Looks like it :)